ui/src/app/components/runs/run-detail/run-detail.component.ts

```diff
@@ -28,6 +28,7 @@ import { Subscription } from 'rxjs';
 })
 export class RunDetailComponent implements OnInit, OnDestroy {
   @Input('dagRunId') dagRunId: number;
+  @Input() refreshSubject: Subject<boolean> = new Subject<boolean>();
```

Comment:
Looks fine, but I noticed unnecessary backend calls, every time you open detail it creates new subscription, so it if you open and close details you will end up with 10 calls. To fix you can do this:
1. In onRefresh method have only `this.store.dispatch(new GetDagRunDetail(this.dagRunId));`
2. In ngOnInit `this.refreshSubscription = this.refreshSubject.subscribe((response) => {
        if (response) {
          this.onRefresh();
      }
    });` 
3. Do not forget unsubscribe from subscription `!!this.refreshSubscription && this.refreshSubscription.unsubscribe();`

Reply:
changes have been made, thanks 😊  for this.